### PR TITLE
@0x48piraj : Designed Badgeyay Front-End

### LINKED ISSUE #357: Seperate front-end and backend / Create front-end using emberJS
Please seperate front-end and backend. Create a folder for the front-end. Use ember JS as front-end technology.

### PR DESCRIPTION
**Pull request by 0x48piraj**
# Fixes #357

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] I have added necessary documentation (if appropriate)

## Changes proposed in this pull request:

- **_Made a fully responsive ember-js based frontend :heart_decoration:_** 
- **_Custom Radio buttons, custom buttons._**
- **_Designed a Loading Screen :tada: :heart:_** 
- **_Implemented a switch to improve UI and to clarify Preview, Generate buttons_**
- **_Designed a Custom Heading_**
- **_Improvised Upload button to make it more responsive._**
- **_Added a fully responsive native color picker *(light weight, native support, greater control)* :1st_place_medal:_** 
- **_Many small tweaks to make the generator look good. :100:_** 

### Bug Fixes : 

- **_Fixed the text-area bug, that is present in our main app other front-ends. :1st_place_medal:_** 

badgeyay_bug1

### Deployed App : 

**_Deploy Link : https://piyushyay-frontend.herokuapp.com/_** :tada: 

### Live Demo : 
badgeyay_demo

### WIP
-  **Adding Documentation/info links.**
- **Improving UI.**
---

### After GCI :

- **Use _https://www.npmjs.com/package/handlebars-helper-css_ to load css.**

- **Customize upload button**

- **Custom Text-Area** 

- **Help in connecting the Back-End**

- **Combining/Compressing CSS and JS Files**

- **Design a logo**

### CODE REVIEW

**Comment by 0x48piraj** [comment]
Ping @fossasia/badgeyay-admins @fossasia/badgeyay-devs @mariobehling @hpdang :heart_decoration: 

> **The Travis-CI is failing not because of my changes.
Ref :** _**https://travis-ci.org/fossasia/badgeyay/builds/327637539**_

**Comment by anshumanv** [comment]
Frontend design looks neat! :+1:

**Comment by 0x48piraj** [comment]
@akshatnitd I'm waiting for reviews, also the `development` repo is constantly changing, I will update it after some decision is made. Am I thinking the correct way ? Please correct me. :relaxed:

File: 0x48piraj/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# PiyushYay Front-End !
```
Comment (vaibhavsingh97) [inline 0x48piraj/README.md]: Please add how one can run your ember app

File: 0x48piraj/app/components/badge-front.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        /*  Show the json task*/
+        this.$(".config-json").css({
+          'display' : 'block'
+        });
+        this.$(".custom-text").css({
+          'display' : 'none'
+        });
+      }
+    }
+
```
Comment (vaibhavsingh97) [inline 0x48piraj/app/components/badge-front.js]: remove balnk lines

File: 0x48piraj/app/styles/app.css [hunk 3]
```diff
@@ -0,0 +1,150 @@
+@import url(https://fonts.googleapis.com/css?family=Bungee);
```
Comment (vaibhavsingh97) [inline 0x48piraj/app/styles/app.css]: Are you using all fonts? also, import should not be used. It should be avoided. Read more about it here: https://www.giftofspeed.com/avoid-using-css-import/ and https://www.stevesouders.com/blog/2009/04/09/dont-use-import/

File: 0x48piraj/app/styles/app.css [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+.csv-upload p, .manual-data p, .png-background p, .default-image-background p, .custom-background p, .custom-text p, .config-json p{
+  margin-top: 0;
+}
+
+.manual-data  textarea {
+  width: 90%;
+  height: 50px;
+}
+
```
Comment (vaibhavsingh97) [inline 0x48piraj/app/styles/app.css]: inconsistent spacing

File: 0x48piraj/app/styles/app.css [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+.style {
+  text-shadow: 1px 2px 0px #ebebeb,
+               0 0 0 black,
+               -1px 1px 0 #ebebeb,
+               5px 5px 0 #333;
+  
+  color: rgba(64, 65, 63, 1.0000);
+}
+
```
Comment (vaibhavsingh97) [inline 0x48piraj/app/styles/app.css]: same as above

File: 0x48piraj/app/templates/components/badge-front.hbs [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+<div id="app">
+  <form action="">
+    <div class="pad">
+
+{{!-- Data Source --}}
+  <div style="width:100%" id="r8-balloon-radio" class="container">
+    <p style="font-family: 'Bungee';">Choose your Data Source</p><p style="font-family: 'Bangers';"><br><br>
+    <ul>
+        Add CSV file&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<li><input id="csv" class="radio r8-radio-float" type="radio" name="data-source" value="0" onclick={{action 'datasourceChanged' 'csv'}} /></li><br>
```
Comment (vaibhavsingh97) [inline 0x48piraj/app/templates/components/badge-front.hbs]: Don't use `&nbsp;` like this. Use css properties.

File: 0x48piraj/app/templates/components/badge-front.hbs [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+        <textarea style="resize: none;width:100%" name="names-csv" id="names-csv" placeholder="Eg. Hong Phuc,Dang,@hpdang,FOSSASIA "></textarea>
+</section>
+
+
+{{!-- Badge Background --}}
+
+<div style="width:100%" id="r8-balloon-radio" class="container">
+    <p style="font-family: 'Bungee';">Choose your Badge Background</p><p style="font-family: 'Bangers';"><br><br>
+    <ul>
+        Add PNG file&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<li><input class="radio r8-radio-float" type="radio" value="0" id="png-image" name="badge-background" onclick={{action 'backgroundsourceChanged' 'png'}} /></li><br>
```
Comment (vaibhavsingh97) [inline 0x48piraj/app/templates/components/badge-front.hbs]: same

File: 0x48piraj/dist/button/css/style.css [hunk 8]
```diff
@@ -0,0 +1,59 @@
+.draw-border {
```
Comment (vaibhavsingh97) [inline 0x48piraj/dist/button/css/style.css]: Please use one space between two classes

Comment (vaibhavsingh97) [reply 0x48piraj/dist/button/css/style.css]: Please reformat this file

File: 0x48piraj/dist/css/style.css [hunk 9]
```diff
@@ -0,0 +1,292 @@
+@import "https://s3-us-west-2.amazonaws.com/s.cdpn.io/1251/bootstrap.min.css";
+@import "https://s3-us-west-2.amazonaws.com/s.cdpn.io/1251/r8-style.css";
+* {
```
Comment (vaibhavsingh97) [inline 0x48piraj/dist/css/style.css]: same

File: 0x48piraj/dist/loading.html [hunk 10]
```diff
@@ -0,0 +1,106 @@
+<!DOCTYPE html>
```
Comment (vaibhavsingh97) [inline 0x48piraj/dist/loading.html]: purpose of this file?

Comment (0x48piraj) [reply 0x48piraj/dist/loading.html]: While our documents get ready, we can embed this design until then.

**Comment by duttaditya18** [comment]
I really think this should be merged as soon as possible. Small problems can be taken care of later but this contributes to the project in a huge way and delaying it would just make merging more and more complex.

**Comment by vaibhavsingh97** [comment]
@0x48piraj Any update?

**Comment by 0x48piraj** [comment]
Give me some time, busy in making voting website for meilix, will do before 20th.

**Comment by mariobehling** [comment]
Any development work should go to the dev branch. Please avoid different UIs and keep the existing UI for now.

**Comment by 0x48piraj** [comment]
This was a GCI Task.

**Pull request closed without merging**